### PR TITLE
PROW-2106: Added allocation name field

### PR DESCRIPTION
**Pull request by vanhaxl**
Add name field for Allocation

### CODE REVIEW

File: proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java [hunk 1]
```diff
@@ -37,13 +40,24 @@ public Allocation(@Nullable final String rule,
         this.id = Strings.nullToEmpty(id);
```
Comment (stevenchen-indeed) [inline proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java:40]: should it calls the new constructor with id and name?

Comment (vanhaxl) [reply proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java:40]: Yes, it should call the new constructor with the id and name. Still keep some previous constructors, so if the client doesn't send the name field, it still works as well

Comment (stevenchen-indeed) [reply proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java:40]: sounds good.

Just to clarify, i was asking whether this method should become this(rules, changes, id, Strings.EMPTY)?

Comment (vanhaxl) [reply proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java:40]: Good point. I think so. It should become this (rules, range, id, "")
I'll make the change

Comment (vanhaxl) [reply proctor-common/src/main/java/com/indeed/proctor/common/model/Allocation.java]: I updated

**Review by zacharygoodwin — APPROVED** [review]

**Pull request merged**